pyemma/msm/estimators/maximum_likelihood_msm.py

```diff
@@ -1361,3 +1361,498 @@ def OOM_sigma(self):
+        dmhat_old = self._dmhat.copy()
+        #slopesum is the sum-of-slopes it is used as an additional ad hoc convergence criterion
+        old_slopesum = _np.abs(self._S).sum()
+        slopesum = old_slopesum+1
+        while((self._ll_old>_ll_new) or (_np.any(self.pihat<1e-12)) or slopesum>old_slopesum):
+            self._update_pihat()
+            self._update_G()
+            # Lagrange slope calculation
+            dl = 2.*(frac*self._G*self.w[:, None]*self._S[:, None]).sum(axis=0)
+            #update Lagrange multipliers
+            self.lagrange = l_old - frac*dl 
+            self._update_pihat()
+            # a number of sanity checks 
+            while(_np.any(self.pihat<1e-12)):
+                frac = frac*0.5
+                self.lagrange = l_old - frac*dl 
+                self._update_pihat()
+
+            self.lagrange = l_old - frac*dl 
+            self._update_pihat()
+            self._update_mhat()
+            self._update_Q()
+            self._update_X_and_pi()
+
+            P = self.X / self.pi[:, None]
+            _ll_new = self._log_likelihood_biased(self._C_active, P, self.m, self.mhat, self.w)
+            # decrease slope in Lagrange space (only used if loop is repeated, e.g. if sanity checks fail)
+            frac = frac*0.1
+
+            if frac<1e-12:
+                self.logger.info("Small gradient fraction")
+                break
+
+            self._dmhat = self.mhat - mhat_old
+            self._ll_old = float(_ll_new)
+            slopesum = _np.abs(self._S).sum()
+
+        self._lls.append(_ll_new)
+
+    def _estimate(self, dtrajs):
+        if self.E is None or self.w is None or self.m is None:
+            raise ValueError("E, w or m was not specified. Stopping.") 
+        die = False
+
+        # get trajectory counts. This sets _C_full and _nstates_full
+        dtrajstats = self._get_dtraj_stats(dtrajs)
+        self._C_full = dtrajstats.count_matrix()  # full count matrix
+        self._nstates_full = self._C_full.shape[0]  # number of states
+
+        # set active set. This is at the same time a mapping from active to full
+        if self.connectivity == 'largest':
+            # statdist not given - full connectivity on all states
+            self.active_set = dtrajstats.largest_connected_set
+        else:
+            # for 'None' and 'all' all visited states are active
+            self.active_set = dtrajstats.visited_set
+
+        # FIXME: setting is_estimated before so that we can start using the parameters just set, but this is not clean!
+        # is estimated
+        self._is_estimated = True
+
+        # if active set is empty, we can't do anything.
+        if _np.size(self.active_set) == 0:
+            raise RuntimeError('Active set is empty. Cannot estimate AMM.')
+
+        # active count matrix and number of states
+        self._C_active = dtrajstats.count_matrix(subset=self.active_set)
+        self._nstates = self._C_active.shape[0]
+        
```

Comment:
Not using PyCharm so I'm not sure what you are referring to. I'm guessing line 1683 should go?